librosa/core/audio.py

```diff
@@ -353,6 +353,10 @@ def stream(path, block_length, frame_length, hop_length,
     # Get the sample rate from the file info
     sr = sf.info(path).samplerate
 
+    # If the input is a file handle, rewind its read position after `sf.info`
+    if not isinstance(path, str) and not isinstance(path, int):
+        path.seek(0)
+
```

Comment:
I wonder if there's a more direct way to detect if seeking is supported here?  Checking for `str` type is brittle (esp in py2, where str and unicode may not be the same).

It might be more robust to just do this seek if `hasattr(path, 'seek')`